### PR TITLE
[WIP] #122 Goby Lang HTTP API

### PR DESCRIPTION
**Pull request by Maxwell-Alexius**

### CODE REVIEW

File: lib/net/http/request.gb [hunk 1]
```diff
@@ -1,7 +1,19 @@
 module Net
   class HTTP
     class Request
-      attr_accessor :body, :method, :path, :url, :host
+      attr_accessor :method, :protocol, :body, :content_length, :transfer_encoding, :host, :path, :url
+      attr_reader   :header
```
Comment (st0012) [inline lib/net/http/request.gb]: This should be `headers`

File: vm/instruction.go [hunk 2]
```diff
@@ -453,6 +453,8 @@ func (vm *VM) initObjectFromGoType(value interface{}) Object {
 	switch v := value.(type) {
 	case int:
 		return vm.initIntegerObject(int(v))
+	case int64:
```
Comment (st0012) [inline vm/instruction.go]: I think we'll write an object mapper very soon...

File: vm/simple_server.go [hunk 3]
```diff
@@ -20,11 +20,15 @@ type response struct {
...
-	Path   string
-	Host   string
+	Method   string
+	Body     string
+	URL      string
+	Path     string
+	Host     string
+	Protocol string
+	Header   map[string][]string
+	Content_Length int64
```
Comment (st0012) [inline vm/simple_server.go]: Please fix this

File: vm/instruction.go (hunk 2 above)
Comment (Maxwell-Alexius) [inline vm/instruction.go]: Yeah, and it is not a good practice to convert `int64` back to `int` type which might cause value being truncated.

Comment (Maxwell-Alexius) [inline vm/instruction.go]: But for now, the object mapper deserves another issue or PR to discuss about it @@

File: lib/net/http/request.gb [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
+        end
+        @headers[key] = value
+      end
+
+      def get_header(key)
+        return @headers[key]
+      end
+
+      def delete_header(key)
+      end
```
Comment (Maxwell-Alexius) [inline lib/net/http/request.gb]: The `delete_header` method should be use the `Hash#delete` method which isn't implemented yet but stated in #247

File: vm/simple_server.go [hunk 5]
```diff
@@ -20,11 +20,15 @@ type response struct {
...
 
 type request struct {
-	Method string
-	Body   string
-	URL    string
-	Path   string
-	Host   string
+	Method           string
+	Body             string
+	Url              string
```
Comment (Maxwell-Alexius) [inline vm/simple_server.go]: I'll fix this @@

File: lib/net/http/request.gb [hunk 6]
```diff
@@ -1,7 +1,22 @@
 module Net
   class HTTP
     class Request
-      attr_accessor :body, :method, :path, :url, :host
+      attr_accessor :method, :protocol, :body, :content_length, :transfer_encoding, :host, :path, :url
+      attr_reader   :headers
+
+      def set_header(key, value)
+        if @headers == nil
```
Comment (Maxwell-Alexius) [inline lib/net/http/request.gb]: Is there any better type checking?

File: vm/simple_server_test.go [hunk 7]
```diff
@@ -78,6 +78,22 @@ func TestInitRequest(t *testing.T) {
...
+			11, // Length of the body: "Hello World"
+		},
+		//{
+		//	"@transfer_encoding",
+		//	0,
+		//},
+		//{
+		//	"@headers",
+		//	initHashObject,
+		//},
```
Comment (Maxwell-Alexius) [inline vm/simple_server_test.go]: Is there any better idea to test this `@transfer_encoding` or `@headers` thing?

Comment (st0012) [inline vm/simple_server_test.go]: I think hash attributes like header should have a map<->hash matching function

**Pull request merged**